code-2.java

```diff
@@ -1,153 +0,0 @@
-import java.util.*;
-
-public class Main {
-
-    public static void main(String[] args) {
-
-        // Setup number of guesses allowed
-        int guessesRemaining = 3;
-
-        // Create scanner object to read user input
-        Scanner input = new Scanner(System.in);
-
-        // Generate game word
-        char[] GAME_WORD = generateGameWord();
-        char[] gameBoard = GAME_WORD.clone();
-
-        // Set game board to all _
-        for(int itr=0; itr < gameBoard.length; itr++)
-            gameBoard[itr] = "_".charAt(0);
-
-
-        // Boolean to track state of game
-        boolean game_over = false;
-        
-        // Print initial status
-        System.out.println("Welcome to hang man. Here is your word to guess");
-        printGameBoard(gameBoard);
-        
-        // Main game loop, use game_over boolean to be more readable than infinite while with breaks
-        while(!game_over){
-
-            System.out.println("Please enter a single valid character to guess");
-            String letter = input.next();
-
-            // Validate input is a singe char
-            while(!validInput(letter)){
-                System.out.println("Error: Invalid input provided. Please enter a single letter to guess");
-                letter = input.next();
-            }
-
-            // Convert char to lower case so we can accept upper or lower case guesses
-            letter = letter.toLowerCase(Locale.ROOT);
-
-            System.out.println("You guessed " + letter);
-
-            // Process the guessed letter
-            if(!guessLetter(letter, gameBoard, GAME_WORD)){
-                System.out.println("You guessed incorrectly. Minus one guess");
-                guessesRemaining--;
-            }
-
-            // Print state of game board
-            printGameBoard(gameBoard);
-            System.out.println("Lives remaining: " + String.valueOf(guessesRemaining));
-
-            // Check win and loss condition
-            if(guessesRemaining <= 0){
-                System.out.println("GAME OVER. Out of lives");
-                game_over = true;
-                break;

```

Comment:
Isn't this break redundant. Boolean variables are often not the best way to express something. In this case you can probably just make the larger while loop condition as `while(guessesRemaining > 0 && String.valueOf(gameBoard).equals(String.valueOf(GAME_WORD))) or something. The latter bit seems kind of verbose though, might be a more concise way to write that but its also java lol